### DISCUSSION TITLE
Re-establish realtime connection after restart

### DISCUSSION THREAD

**Discussion by magnusskare** [pocketbase/pocketbase#4661] [Q&A] [answered]
I'm using Pockethost, and sometimes I have to restart the application. How long does it take to re-establish a realtime connection to someone already viewing the app? Is there a way to force the re-connect?

For some context, I'm using hooks to delete/create some records on startup.

```
onAfterBootstrap((e) => {
    let existingNotes = $app.dao().findRecordsByFilter('notes', "title == 'Temporary'", '-created', 50);
    existingNotes.forEach(item => {
    try {
      let record = $app.dao().findRecordById('notes', item.id);
      $app.dao().deleteRecord(record);
    } catch (err) {
      $app.logger().debug("missing record or failed to save", "error", err);
    }
  });
})
````

I've attempted to use ```$app.dao().deleteRecord(record)``` and as well as ```$app.dao().saveRecord(record)``` inside ```onAfterBootstrap```, which [I thought](https://github.com/pocketbase/pocketbase/discussions/3695) would trigger realtime events. Does it not trigger on ```onAfterBootstrap```? The linked answer was related to ```onRealtimeDisconnectRequest```, which might be different.

**Comment by ganigeorgiev** [comment, marked as answer]
> How long does it take to re-establish a realtime connection to someone already viewing the app?  Is there a way to force the re-connect?

The reconnect is automatically done by the SDKs. The JS SDK tries in `[200, 300, 500, 1000, 1200, 1500, 2000, 2000, 2000, ...]` ms intervals.

`onAfterBootstrap` is fired before the web server start so the above will not trigger the realtime events.

We currently don't have an event that will fire after the web server start, aka. "after serve" (_I'll consider it with the [refactored hooks](https://github.com/pocketbase/pocketbase/discussions/4355)_).


If you want to listen to connect/reconnect events on the client you can subsribe to the `PB_CONNECT` topic:

```js
pb.realtime.subscribe("PB_CONNECT", callback)
```

**Reply by magnusskare** [reply]
Thank you for the clarification. This helped me get a working MVP of this going, as I can manually check for updates/changed on ```PB_CONNECT``` with the callback. Combined with the Page Visibility API, we can also prevent unnecessary database queries when the tab is inactive, and instead queue the manual check for when they return to the browser tab.

Thanks!

I'm assuming I can just unsubscribe by calling ```pb.realtime.unsubscribe('PB_CONNECT')```.
